components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -421,6 +459,41 @@ public void updateConsumerApplication(OAuthAppDO oauthAppDO) throws IdentityOAut
                 if (log.isDebugEnabled()) {
                     log.debug("No. of records updated for updating consumer application. : " + count);
                 }
+
+                if (OAuth2ServiceComponentHolder.isAudienceEnabled()) {
+                    String[] audiences = oauthAppDO.getAudiences();
+                    HashSet<String> newAudiences = new HashSet<>(Arrays.asList(audiences));
```

Comment:
Use Set interface when assigning variable

Reply:
Cannot use Set interface here since the latter part of the logic utilizes HashSet specific methods.